src/timeMachine/components/dynamicFluxFunctionsToolbar/FluxFunctionsToolbar.tsx

```diff
@@ -8,56 +9,105 @@ import {DapperScrollbars} from '@influxdata/clockface'
 import ErrorBoundary from 'src/shared/components/ErrorBoundary'
 import ToolbarFunction from 'src/timeMachine/components/dynamicFluxFunctionsToolbar/ToolbarFunction'
 
-// Constants
-import {FLUX_FUNCTIONS} from 'src/shared/constants/fluxFunctions'
+import {SpinnerContainer, TechnoSpinner} from '@influxdata/clockface'
+
+import {Fluxdocs} from 'src/client/fluxdocsdRoutes'
+import {getFluxPackages} from 'src/timeMachine/actions/scriptEditorThunks'
 
 // Types
-import {FluxToolbarFunction} from 'src/types'
+import {RemoteDataState} from 'src/types'
+import {AppState} from 'src/types'
 
 interface OwnProps {
-  onInsertFluxFunction: (func: FluxToolbarFunction) => void
+  onInsertFluxFunction: (func) => void
+}
+
+interface DispatchProps {
+  getFluxPackages: () => void
 }
 
-const DynamicFluxFunctionsToolbar: FC<OwnProps> = (props: OwnProps) => {
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & OwnProps & DispatchProps
+
+const DynamicFluxFunctionsToolbar: FC<Props> = (props: Props) => {
   const [searchTerm, setSearchTerm] = useState('')
+  const [fluxServiceError, setFluxServiceError] = useState<RemoteDataState>(
+    RemoteDataState.NotStarted
+  )
 
   const handleSearch = (searchTerm: string): void => {
     setSearchTerm(searchTerm)
   }
 
+  useEffect(() => {
+    const getFluxFuncs = async () => {
+      try {
+        setFluxServiceError(RemoteDataState.Loading)
+
+        if (props.fluxFunctions.length === 0) {
+          await props.getFluxPackages()
+          setFluxServiceError(RemoteDataState.Done)
+        }
+        setFluxServiceError(RemoteDataState.Done)
+      } catch (err) {
+        console.error(err)
+        setFluxServiceError(RemoteDataState.Error)
+      }
+    }
+    getFluxFuncs()
+  }, [])
+
   const handleClickFunction = useCallback(
-    (func: FluxToolbarFunction) => {
+    (func: Fluxdocs) => {
       props.onInsertFluxFunction(func)
     },
     [props.onInsertFluxFunction]
```

Comment:
This was already in place and not changed, but i just noticed it. We should make an improvement here.

eslint is complaining here because `props` is a dependency. To get around this, destructure the `onInsertFluxFunction` from `props` near the top and then use just `onInsertFluxFunction` in here, then eslint will stop complaining.

Reply:
of course! It looks unnecessarily messy when it's not destructured.